Automattic/wp-calypso
# Gutenberg: Remove feature flags and simplify flows

#### Changes proposed in this Pull Request

* Removed the feature flags `gutenberg` and `gutenberg/opt-in` since they all have the same value in all environments and we don't have any plan to disable or revert the block editor.
* Simplified the selectors `isGutenbergEnabled`, `isGutenframeEnabled` (previously `isGutenlypsoEnabled`) and `isCalypsoifyGutenbergEnabled` to better handle the different sites/editors combinations:

**Simple sites**

| Selected editor | Loaded editor |
|---|---|
| `classic` | Calypso classic editor |
| `gutenberg` | Iframed blocked editor |

**JP/AT sites**

| WP version | Classic Editor plugin | Selected editor | JP version | SSL | Loaded editor |
|---|---|---|---|---|---|
| <5.0 | N/A | N/A | N/A | N/A | Calypso classic editor |
| >=5.0 | Yes | N/A | N/A | N/A | Calypso classic editor |
| >=5.0 | No | `classic` | N/A | N/A | Calypso classic editor |
| >=5.0 | No | `gutenberg` | <7.3 | N/A | Calypsoified WP Admin block editor |
| >=5.0 | No | `gutenberg` | >7.3 | No | Calypsoified WP Admin block editor |
| >=5.0 | No | `gutenberg` | >7.3 | Yes | Iframed blocked editor |

I noticed that the selectors were not able to detect if the Classic Editor plugin was installed because we didn't have any plugin stored on the state. I realized this was due to the lack of any dispatched action fetching the plugin and probably the cause of the issue described on https://github.com/orgs/Automattic/projects/88#card-20203410. This why I included a `QueryJetpackPlugins` query component in the `Layout` component.

#### Testing instructions



* Make sure that the Gutenberg opt-in/opt-out flows still work as expected.
* Make sure that the Calypso loads the expected editor according to the combinations table above.

Fixes #32626
